sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs

```diff
@@ -126,26 +129,12 @@ public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientDiagnostics,
 
         private async Task<FetchResult> FetchAsync(string dtmi, CancellationToken cancellationToken)
         {
-            try
-            {
-                return await _modelFetcher.FetchAsync(dtmi, _repositoryUri, cancellationToken).ConfigureAwait(false);
-            }
-            catch (Exception ex)
-            {
-                throw new ResolverException(dtmi, ex.Message, ex);
-            }
+            return await _modelFetcher.FetchAsync(dtmi, _repositoryUri, cancellationToken).ConfigureAwait(false);
```

Comment:
for simple methods that make 1 async all, you can instead choose to:

1. Remove the `async` keyword in the method signature
2. Remove the `await` keyword on the method call in the return statement
3. Remove the `.ConfigureAwait(false)` call on that method.

What happens if the Task is being returned to the caller, who will then await it. Supposedly it reduces some overhead that comes with async/await.